### DISCUSSION TITLE
The method `this.emitFieldValue` not works for belongsTo

### DISCUSSION THREAD

**Discussion by leandrogehlen** [laravel/nova-issues#5455] [Q&A] [answered]
- Laravel Version:  9.52.5
- Nova Version: 4.22.2
- PHP Version: 8.1.2
- Database Driver & Version: mysql

### Description:

Building a new custom field component i need update `BelongsTo` relation via vue component.

using the `this.emitFieldValue` works fine with simple field (Text, Number ...) but for `BelongsTo` component 
when set key value (`this.emitFieldValue('city', 1)`) should be displayed the city name.

https://user-images.githubusercontent.com/1750751/230185666-3f1e84cd-d7bf-4690-8fbb-c9614ae22511.mp4

**Comment by crynobone** [comment]
`BelongsTo` field doesn't support this usage, you should use Dependent Field to trigger update to `BelongsTo` field.

**Reply by leandrogehlen** [reply]
Dependent fields has the same issue.

I have tried
```php
BelongsTo::make(__('City'), 'city', City::class)
    ->searchable()
    ->withoutTrashed()
    ->rules('required')
    ->dependsOn(['postcode'], function (BelongsTo $field, NovaRequest $request, FormData $formData) {
       $field->value = 3550308;
   }),
```
But it is not works.

I believe that the problem is the same for both scenarios (`dependent fields` and `emitFieldValue`. I think that is a necessary feature and is a simple modification. It's a little strange that `emitFieldValue` works for all other fields except `BelongsTo`. This feature would be very helpful building more complex routines and custom components.

**Reply by crynobone** [reply]
https://github.com/laravel/nova-dusk-suite/blob/e7f23da12f05d15dd46ad5fc524afa049bbda1bd/app/Nova/Post.php#L64-L81

**Reply by leandrogehlen** [reply]
Ok, it works for `default` because it's a creation operation
But when is editing a resource the issue happens

I have detected the problem in `initializeComponent` method,
line: `this.selectedResourceId = this.currentField.belongsToId`

```js
initializeComponent() {
   this.withTrashed = false  

  this.selectedResourceId = this.currentField.value // currentField.value is correct

  if (this.editingExistingResource) {
     // If a user is editing an existing resource with this relation
     // we'll have a belongsToId on the field, and we should prefill
     // that resource in this field
     this.initializingWithExistingResource = true
      this.selectedResourceId = this.currentField.belongsToId // this change the new value defined in dependsOn closure
   } else if (this.creatingViaRelatedResource) {
      ...
```

**Reply by crynobone** [reply, marked as answer]
I believe this limitation has been fixed in the latest Laravel Nova 4 release.
